sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -127,13 +128,19 @@ trait InvokeLike extends Expression with NonSQLExpression {
       arguments: Seq[Expression],
       input: InternalRow,
       dataType: DataType): Any = {
-    val args = arguments.map(e => e.eval(input).asInstanceOf[Object])
-    if (needNullCheck && args.exists(_ == null)) {
+    var i = 0
+    val len = arguments.length
+    while (i < len) {
+      val e = arguments(i)
+      evaluatedArgs(i) = e.eval(input).asInstanceOf[Object]
+      i += 1
+    }
+    if (needNullCheck && evaluatedArgs.exists(_ == null)) {
```

Comment:
nit: my IDE suggests `.exists(_ == null)` -> `.contains(null)`

Reply:
the `exists` part is not related to this PR but I'm happy to change it :)